src/ad-cid.js

```diff
@@ -29,13 +29,28 @@ export function getAdCid(adElement) {
   if (!config || !config.clientIdScope) {
     return Promise.resolve();
   }
-  const cidPromise = cidForDocOrNull(adElement.getAmpDoc()).then(cidService => {
+  return getClientScopedAdCid(adElement.getAmpDoc(), adElement.win,
+                              config.clientIdScope, config.clientIdCookieName);
+}
+
+/**
+ * @param {!./service/ampdoc-impl.AmpDoc} doc
+ * @param {!Window} win
+ * @param {!string} clientIdScope
+ * @param {string=} opt_clientIdCookieName
+ * @return {!Promise<string|undefined>} A promise for a CID or undefined if
+ *     - the ad network does not request one or
+ *     - `amp-analytics` which provides the CID service was not installed.
+ */
+export function getClientScopedAdCid(
+    doc, win, clientIdScope, opt_clientIdCookieName) {
+  const cidPromise = cidForDocOrNull(doc).then(cidService => {
     if (!cidService) {
       return;
     }
     return cidService.get({
-      scope: dev().assertString(config.clientIdScope),
-      cookieName: config.clientIdCookieName,
+      scope: dev().assertString(clientIdScope),
+      cookieName: opt_clientIdCookieName || undefined,
```

Comment:
Why do you need || undefined here?  You're guarding against a null value for cookieName?